sycl/doc/CompilerAndRuntimeDesign.md

```diff
@@ -819,29 +820,27 @@ Depending on the compiler mode `multi_ptr` will either annotate internal data
 with address space attribute or not.
 
 To utilize existing clang's functionality, we re-use following OpenCL address
-space attributes in SYCL mode with modified semantic rules and unmodified parser
-and IR generation components.
-
-``` C++
-  __attribute__((opencl_global))
-  __attribute__((opencl_local))
-  __attribute__((opencl_private))
-```
-
-The main difference in Sema component with OpenCL mode is that SYCL mode
-(similar to other single-source GPU programming modes like OpenMP/CUDA/HIP)
-keeps "default" address space for the declaration without explicit address space
-attribute annotations. This keeps the type system for the code shared between
-the host and device semantically-equivalent for both compilers: regular C++ host
-compiler and SYCL compiler.
-
-SYCL mode allows conversion from annotated pointer to unannotated pointer to
-enable integration of accelerated code with standard C++ code.
-
-> **NOTE:** casting unannotated pointer to address-space annotated pointer must
-be done with care. If address space of annotated pointer doesn't match the
-address space of the allocation unannotated pointer is pointing to, the behavior
-of cast operation is undefined.
+space attributes in SYCL mode: `__attribute__((opencl_global))`,
+`__attribute__((opencl_global_host))`, `__attribute__((opencl_global_device))`,
+`__attribute__((opencl_local))`, `__attribute__((opencl_private))` and
+`__attribute__((opencl_constant))`.
+
```

Comment:
SYCL-2020 defines these:
``` C++
3 enum class address_space : int {
4 global_space,
5 local_space,
6 constant_space, // Deprecated in SYCL 2020
7 private_space,
8 generic_space,
9 };
```
Should we give a mapping of "opencl" attributes to SYCL semantics? Also why not add macros for users to see the more meaningful name (even though that underneath "opencl" attributes are re-used)?

Reply:
> Should we give a mapping of "opencl" attributes to SYCL semantics? 

Could you clarify what exactly do you mean here? I suppose provided implementation of `multi_ptr` with reference to `DecoratedType` helper class gives that mapping. Do I miss something?

> Also why not add macros for users to see the more meaningful name (even though that underneath "opencl" attributes are re-used)?

SYCL provides explicit pointer aliases (see section 4.7.7.2 of SYCL-1.2.1) for users. Do you think we need additional macros?
Could you provide an example of such macros, please?

Reply:
I mean we should show here in the designed and to the end users too that "opencl_local" translates into SYCL's "address_space::local_space" and so on.

Reply:
I'll add a table with mapping from address space attribute to SYCL address_space enumeration.